src/main/java/noppes/npcs/scripted/ScriptWorld.java

```diff
@@ -135,17 +144,18 @@ public void removeBlock(int x, int y, int z){
 		world.setBlock(x, y, z, Blocks.air);
 	}
 	
-	public IBlock rayCastBlock(Vec3 pos, Vec3 look, int maxDistance) {
-		Vec3 currentPos = pos; int rep = 0;
+	public IBlock rayCastBlock(double[] pos, double[] look, int maxDistance) {
+		if (pos.length != 3 || look.length != 3) return null;
+		Vec3 currentPos = Vec3.createVectorHelper(pos[0], pos[1], pos[2]); int rep = 0;
 		while (rep++ < maxDistance + 10) {
-			currentPos = currentPos.addVector(look.xCoord, look.yCoord, look.zCoord);
+			currentPos = currentPos.addVector(look[0], look[1], look[2]);
 			IBlock block = getBlock((int)currentPos.xCoord, (int)currentPos.yCoord, (int)currentPos.zCoord);
 			//System.out.println("Checking block at ["+(int)currentPos.xCoord+","+(int)currentPos.yCoord+","+(int)currentPos.zCoord+"]");
 			if (block == null) continue;
 			double distance = Math.pow(
-					Math.pow(currentPos.xCoord-pos.xCoord,2)
-					+Math.pow(currentPos.yCoord-pos.yCoord,2)
-					+Math.pow(currentPos.zCoord-pos.zCoord,2)
+					Math.pow(currentPos.xCoord-pos[0],2)
+					+Math.pow(currentPos.yCoord-pos[1],2)
+					+Math.pow(currentPos.zCoord-pos[2],2)
 					, 0.5);
 			//System.out.println("current distance check: "+distance+" on rep "+rep);
 			if (distance > maxDistance) return null;
```

Comment:
If you could implement these with IPos objects as parameters instead of double[], just like you implemented getNearestAir() here, that would be great!

Reply:
But IPos uses integers. It needs to be doubles. Unless I add xDouble yDouble zDouble fields to IPos

Reply:
ended up making one with double[] parameters and another with IPos parameters